test/ui-test/testSuites/suite_messaging/shared/steps/chatSteps.py

```diff
@@ -54,61 +116,78 @@ def step(context):
 
 @When("the user saves changes")
 def step(context):
-    _statusChat.group_chat_edit_save()
-
-@When("the user pressed enter")
+    _statusChat.group_chat_edit_save() 
+    
+@When("the user leaves current chat")
 def step(context):
-    _statusChat.press_enter_in_chat_input()
+    _statusChat.leave_chat()
 
-@Then("user is able to send chat message")
+#########################
+### VERIFICATIONS region:
```

Comment:
Separating the Given/When/Then like this will make things easier to find (even if some of the step definitions will be multiple, such as Given/Then).

What I'm thinking of eventually is that the verifications themselves will be in the step file. For example:

```
@Then("the last chat message contains |any|")
def step(context, expected_last_message):
  actual_last_message = _statusChat.return_last_message_sent     
  
  verify(not is_null(actual_last_message), "Checking last message sent: " + message)
  verify_text_contains(str(actual_last_message), str(expected_last_message))
```

Makes it easy to evaluate the expected/actual data used in the assertions and the method (text contains, equals, etc.), without having to look through the screen/driver layer.

Reply:
That's a good insight. Here a task to address this proposal: https://github.com/status-im/status-desktop/issues/7939

Please add more details if you consider it!